internal/manifests/mutate.go

```diff
@@ -374,14 +374,8 @@ func mutateIssuer(existing, desired *cmv1.Issuer) {
 }
 
 func mutatePodTemplate(existing, desired *corev1.PodTemplateSpec) error {
-	if err := mergeWithOverride(&existing.Labels, desired.Labels); err != nil {
-		return err
-	}
-
-	if err := mergeWithOverride(&existing.Annotations, desired.Annotations); err != nil {
-		return err
-	}
-
```

Comment:
@yuriolisa there was a reason we did this that i'm trying to remember... AH! it was because if the user has an external controller like argocd, OPA, istio which adds labels/annotations we could inadvertently override them causing an infinite cycle. I think it's something in [this pr](https://github.com/open-telemetry/opentelemetry-operator/pull/3324) where we talked about this. @davidhaja may remember better...

Reply:
@jaronoff97 , thank you for the heads-up. I'm just wondering why the PR says that would be changing to `mergeWithOverwriteWithEmptyValue`, but it remains `mergeWithOverride`.

Reply:
One idea would be creating a new field to set which labels and annotations prefixes should be preserved.
```yaml
preservedAnnotationPrefixes:
  - argocd.argoproj.io
  - istio.io
```